cmd/root.go

```diff
@@ -242,13 +251,24 @@ func (r *Root) execute() {
 		Example: rootExamples,
 	}
 	rootCmd.CompletionOptions.DisableDefaultCmd = true
+
 	rootCmd.SetHelpTemplate(rootCustomHelpTemplate)
 
 	rootCmd.PersistentFlags().BoolVar(&debugMode, "debug", false, "Run in debug mode")
 
-	// Manually parse flags to determine debug mode early
+	rootCmd.PersistentFlags().Bool("version", false, "Fetch the latest version")
+
+	// Manually parse flags to determine debug mode and version flag early
 	debugMode = checkForDebugFlag(os.Args[1:])
-	// Now that flags are parsed, set up the l722ogger
+	versionFlag := checkForVersionFlag(os.Args[1:])
```

Comment:
If you have defined a flag in the above code, why do you have to parse again for the version flag?

Reply:
The manual parsing of flags is performed to determine the value of versionFlag variables before setting up the logger and executing the root command. In my opinion it is the correct way to handle this feature. Let me know if there is an alternate or a better way to do this

Reply:
reference: 

https://chromium.googlesource.com/external/github.com/spf13/cobra/+/4fd30b69ee2b62cf3bbecf0a423f8a1ee47f5f24/user_guide.md#version-flag

Reply:
Refactored

Reply:
I don't think you have got the point I am trying make here. Cobra actually has --version inbuilt you have to provide some template to it show the version properly Please go through the reference again.

Reply:
Sorry for the confusion, It is now done.